### PR TITLE
FAQ node api doc for kms systems

### PR DESCRIPTION
**Pull request by parvez301**
JIRA: https://hello-haptik.atlassian.net/browse/PLAT-1428

### CODE REVIEW

File: docs/bot-builder/integrations/additional-apis.md [hunk 1]
```diff
@@ -192,6 +192,136 @@ curl -X GET \
 }
 ```
 
+## FAQ Node Creation APIs for KMS
+We have many clients who uses FAQ bots, So, everytime when we want to change Q&A, a new csv with node name, question and answer in certain format(given below) needs to be uploaded.
```
Comment (medhaanand) [inline docs/bot-builder/integrations/additional-apis.md]: Please elaborate it a little more like on the lines of -
We have many clients who need FAQ bots. Every time when we want to add/replace/change the content of those FAQs, a new CSV with Node name, Question and Answer in a defined format needs to be uploaded. [Format given below]

File: docs/bot-builder/integrations/additional-apis.md [hunk 2]
```diff
@@ -192,6 +192,136 @@ curl -X GET \
...
+
+Although, frequency of this activity varies from business to business but it's a repeatable to smoothen this tedious process we've developed an API which can be integrated with the client KMS(knowledge Management Systems) via any intermediate layer and everytime when data gets changed in KMS, KMS should get call this API to keep data in sync with the haptik systems.
```
Comment (medhaanand) [inline docs/bot-builder/integrations/additional-apis.md]: We have developed an API which can be integrated with the client KMS(Knowledge Management Systems) via any intermediate layer and every time when data gets changed in KMS, it should call this API to keep data in sync with the haptik systems.

File: docs/bot-builder/integrations/additional-apis.md [hunk 3]
```diff
@@ -192,6 +192,136 @@ curl -X GET \
...
+
+This API allows you to create/update/delete faq nodes via a `POST` request to Haptik Platform. 
+
+> Note: The `base-url` will be provided by Haptik at the time of integration.
+
+FAQ Node creation process API consists of two parts:
+1. **Submit the CSV**: To handle the large csv files, this API takes the request data and after validation submit it to celery worker and returns the `task id`, which executes the task asynchronously.
```
Comment (medhaanand) [inline docs/bot-builder/integrations/additional-apis.md]:
```suggestion
1. **Submit the CSV**: To handle the large csv files, this API takes the request data and after backend validation submits it to celery worker and returns the `task id`, which executes the task asynchronously.
```

File: docs/bot-builder/integrations/additional-apis.md [hunk 4]
```diff
@@ -192,6 +192,136 @@ curl -X GET \
...
+2. **Polling API** to fetch the status of submitted `task id`
+
+Example URL: `https://<base-url>/mogambo_api/nodes/faq/create/`
+
+![CSV Format](assets/faq_node_csv_format.png)
+
+```
+1. node_name: Name of the node, node name should be unique
+
+2. question(user says): user says to detect this faq node, multiple user says for single node can be passed as pipe(|)  separated values(ref the sample row in image above)
```
Comment (medhaanand) [inline docs/bot-builder/integrations/additional-apis.md]:
```suggestion
2. Question(User Says): Independent Responses under User Says to detect this FAQ node. Multiple user says for single node can be passed as pipe(|) separated values(refer the sample row in image above)
```

File: docs/bot-builder/integrations/additional-apis.md [hunk 5]
```diff
@@ -192,6 +192,136 @@ curl -X GET \
...
+
+- Authorization - The Authorization header of each HTTP request should be “Bearer” followed by your token which will be provided by Haptik
+- client-id - The client id for your account which will be provided by Haptik
+- Content-Type - multipart/form-data
+
+#### Request Body
+
+```form-data
+"domain_name": "<Name of the Domain/Bot>",
+"csv_file": "csv file",
```
Comment (medhaanand) [inline docs/bot-builder/integrations/additional-apis.md]:
```suggestion
"csv_file": filename.csv,
```

Comment (medhaanand) [reply docs/bot-builder/integrations/additional-apis.md]:
```suggestion
"csv_file": "filename.csv",
```

File: docs/bot-builder/integrations/additional-apis.md [hunk 6]
```diff
@@ -192,6 +192,136 @@ curl -X GET \
...
+
+#### Sample Curl Command to submit csv file and get job id
+```
+curl -X POST \
+  https://<base-url>/mogambo_api/nodes/faq/create/ \
+  -H 'authorization: Bearer <token>' \
+  -H 'client-id: <client-id>' \
+  -H 'content-type: multipart/form-data; 
+  -F 'domain_name=<domain_name>' \
+  -F csv_file=@csv_file.csv
```
Comment (medhaanand) [inline docs/bot-builder/integrations/additional-apis.md]:
```suggestion
  -F csv_file=@filename.csv
```

File: docs/bot-builder/integrations/additional-apis.md [hunk 7]
```diff
@@ -192,6 +192,136 @@ curl -X GET \
...
+```
+
+> Note: This API follows the upsert behavior viz everytime when new csv gets uploaded it make the existing faq nodes for that domain inactive and create new nodes for the rows mentioned in the CSV
```
Comment (medhaanand) [inline docs/bot-builder/integrations/additional-apis.md]:
```suggestion
> Note: This API follows the upsert [UPDATE and INSERT] behaviour i.e. every time when new CSV gets uploaded it makes the existing FAQ nodes for that domain inactive, creates new nodes for the rows mentioned in the CSV and updates Nodes with same names with the new User Says and Bot Says from the CSV. 
```

File: docs/bot-builder/integrations/additional-apis.md [hunk 8]
```diff
@@ -192,6 +192,136 @@ curl -X GET \
...
+
+## Polling API to fetch submitted task Status
```
Comment (medhaanand) [inline docs/bot-builder/integrations/additional-apis.md]: Please put links at relevant places

File: docs/bot-builder/integrations/additional-apis.md (hunk 1 above)
Comment (parvez301) [inline docs/bot-builder/integrations/additional-apis.md]: We have many clients who uses FAQ bots, So, everytime when we want to change Q&A, a new csv with node name, question and answer in certain format(given below) needs to be uploaded.

Comment (parvez301) [inline docs/bot-builder/integrations/additional-apis.md]:
```suggestion
We have many clients who need FAQ bots. Every time when we want to change the content of these FAQs (Q&As), a new CSV with Node name, Question and Answer in a defined format needs to be uploaded. (Format given below)
```

File: docs/bot-builder/integrations/additional-apis.md [hunk 9]
```diff
@@ -192,6 +192,136 @@ curl -X GET \
 }
 ```
 
+## FAQ Node Creation APIs for KMS
+We have many clients who need FAQ bots. Every time when we want to change the content of these FAQs (Q&As), a new CSV with Node name, Question and Answer in a defined format needs to be uploaded. (Format given below)
+
+Although, frequency of this activity varies from business to business but it's a repeatable to smoothen this tedious process we've developed an API which can be integrated with the client KMS(knowledge Management Systems) via any intermediate layer and everytime when data gets changed in KMS, KMS should get call this API to keep data in sync with the haptik systems.
```
Comment (parvez301) [inline docs/bot-builder/integrations/additional-apis.md]:
```suggestion
We have developed an API that can be integrated with the client KMS(Knowledge Management Systems) via any intermediate layer and every time when data gets changed in KMS, it should call this API to keep data in sync with the Haptik systems.
```

**Pull request merged**